modules/core/src/commonMain/kotlin/fr/acinq/lightning/payment/OutgoingPaymentHandler.kt

```diff
@@ -116,10 +116,14 @@ class OutgoingPaymentHandler(val nodeParams: NodeParams, val walletParams: Walle
             logger.error { "contract violation: caller is recycling uuid's" }
             return Failure(request, FinalFailure.InvalidPaymentId.toPaymentFailure())
         }
-        if (db.listLightningOutgoingPayments(request.paymentHash).find { it.status is LightningOutgoingPayment.Status.Succeeded } != null) {
+        if (db.listLightningOutgoingPayments(request.paymentHash).any { it.status is LightningOutgoingPayment.Status.Succeeded }) {
```

Comment:
Should we not also check the status of parts?

That's what we did in Phoenix when applying the same restriction (https://github.com/ACINQ/phoenix/blob/a1dd05d23538f0916cbd1751c826342a100fb4c4/phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/managers/SendManager.kt#L214-L221). Maybe it's not actually useful though.

Reply:
Not only is it redundant, it is also confusing as it assumes impossible combinations are possible.